### PR TITLE
Add MFA support to IAM roles

### PR DESCRIPTION
**Pull request by sid77**
Not sure if the `count` conditional is ok though.

### CODE REVIEW

File: iam/role/main.tf [hunk 1]
```diff
@@ -26,6 +27,13 @@ data "aws_iam_policy_document" "assume_role_policy" {
       type        = "${var.policy_type}"
       identifiers = "${var.policy_identifiers}"
     }
+
+    condition {
+      count    = "${var.mfa ? 1 : 0}"
```
Comment (eredi93) [inline iam/role/main.tf]: this wont work, the count is on a resource level not block level

Comment (eredi93) [inline iam/role/main.tf]: you need to create 2 policies and do an if statement when you pass the policy json

Comment (sid77) [inline iam/role/main.tf]: Code duplication, yay! 🤢

Comment (eredi93) [inline iam/role/main.tf]: there might be a better way. i'm just not aware of it 😢

**Pull request merged**